polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -411,20 +393,9 @@ CodeGenTypes::getFunctionType(const clang::CodeGen::CGFunctionInfo &FI,
     case clang::CodeGen::ABIArgInfo::Indirect: {
       LLVM_DEBUG(llvm::dbgs() << "ArgInfo: ABIArgInfo::Indirect\n");
       assert(NumIRArgs == 1);
-      // indirect arguments are always on the stack, which is alloca addr space.
+      // Indirect arguments are always on the stack, which is alloca addr space.
       mlir::Type MLIRArgTy = getMLIRArgType(DeclArgTy);
-      bool IsSYCLType = mlir::sycl::isSYCLType(MLIRArgTy);
-      if (auto ST = MLIRArgTy.dyn_cast<mlir::LLVM::LLVMStructType>()) {
-        IsSYCLType |= any_of(ST.getBody(), [](auto Element) {
-          return mlir::sycl::isSYCLType(Element);
-        });
-      }
-      if (IsSYCLType)
-        ArgTypes[FirstIRArg] = mlir::MemRefType::get(
-            -1, MLIRArgTy, {}, CGM.getDataLayout().getAllocaAddrSpace());
-      else
-        ArgTypes[FirstIRArg] = LLVM::LLVMPointerType::get(
-            MLIRArgTy, CGM.getDataLayout().getAllocaAddrSpace());
+      ArgTypes[FirstIRArg] = getPointerOrMemRefType(MLIRArgTy);
```

Comment:
Need to pass the address space **explicitly** here so that there are no assumptions on the caller side that something is happening in the function itself. The address space to use is  CGM.getDataLayout().getAllocaAddrSpace()